spring-integration-jms/src/main/java/org/springframework/integration/jms/PollableJmsChannel.java

```diff
@@ -65,6 +100,10 @@ public void setMessageSelector(String messageSelector) {
 			if (object == null) {
 				return null;
 			}
+			if (isCountsEnabled()) {
+				getMetrics().afterReceive();
+				counted = true;
```

Comment:
Why is it here, rather than outside of `if`, like in the `PollableAmqpChannel` ,